src/handler.ts

```diff
@@ -13,50 +13,44 @@ export const handler: Handler = async (event: SQSEvent): Promise<SQSBatchRespons
     logger.error('ERROR: event is not defined.');
     throw new Error('Event is empty');
   }
-
-  const altPromiseArray = event.Records.map((sqsRecord) => {
-    const request = JSON.parse(sqsRecord.body) as Request;
-
-    const document: DocumentModel = getDocumentFromRequest(request);
-    return generateAndUpload(document, request);
-  });
-
-  const results = await Promise.allSettled(altPromiseArray);
-  const ids = results
-    .map((result, index) => (result.status === 'fulfilled' ? null : event.Records[index].messageId))
-    .filter((item) => item !== null);
-  return {
-    batchItemFailures: ids.map((id) => ({ itemIdentifier: id })),
-  };
+  const batchItemFailures: SQSBatchItemFailure[] = [];
+  // eslint-disable-next-line no-restricted-syntax
+  for (const sqsRecord of event.Records) {
+    try {
+      const request = JSON.parse(sqsRecord.body) as Request;
+      const document: DocumentModel = getDocumentFromRequest(request);
+      // eslint-disable-next-line no-await-in-loop
+      await generateAndUpload(document, request);
+    } catch (error) {
+      console.error(error);
+      batchItemFailures.push({ itemIdentifier: sqsRecord.messageId });
```

Comment:
Is there any chance you can add a test to validate the below please?

When a batch of 2 or more records is processed, if one of those falls over, then only the one that failed is added to the batchItemFailures array.

Reply:
Only adding the failures to batchItemFailures - is that not the purpose?

Reply:
Yes exactly, can you create a test that validates that. It looks like there is coverage for the catch block, when an error is thrown, but can't see a test for the scenario above.